src/elements/content-sidebar/activity-feed/task-form/TaskForm.js

```diff
@@ -61,17 +62,36 @@ type Props = TaskFormProps & TaskFormConsumerProps & InjectIntlProvidedProps;
 type TaskFormFieldName = 'taskName' | 'taskAssignees' | 'taskDueDate';
 
 type State = {|
+    addedAssignees: SelectorItems,
```

Comment:
To clean up the add/remove:

- change `state.approvers` type to `Array<TaskCollabAssignee>`
- do the transform to the PillSelector format inside of render
- remove `removedAssignees`, `addedAssignees` state; add/remove from `state.approvers` directly
- do the computation for the added and removed assignees on submit by comparing to the props